include/VectorGraph.h

```diff
@@ -0,0 +1,739 @@
+	//float processLineTypeSine(float xIn, float valA, float valB, float fadeInStartLoc);
+	std::vector<float> processLineTypeArraySine(std::vector<float>* xArray, unsigned int startLoc, unsigned int endLoc,
+		float valA, float valB, float fadeInStartLoc);
+	// curve: phase
+	//float processLineTypeSineB(float xIn, float valA, float valB, float curve, float fadeInStartLoc);
+	std::vector<float> processLineTypeArraySineB(std::vector<float>* xArray, unsigned int startLoc, unsigned int endLoc,
+		float valA, float valB, float curve, float fadeInStartLoc);
+	// valA: amp, valB: x coord, curve: width
+	//float processLineTypePeak(float xIn, float valA, float valB, float curve, float fadeInStartLoc);
+	std::vector<float> processLineTypeArrayPeak(std::vector<float>* xArray, unsigned int startLoc, unsigned int endLoc,
+		float valA, float valB, float curve, float fadeInStartLoc);
+	// y: calculate steps from, valA: y count, valB: curve
+	//float processLineTypeSteps(float xIn, float yIn, float valA, float valBIn, float fadeInStartLoc);
+	std::vector<float> processLineTypeArraySteps(std::vector<float>* xArray, unsigned int startLoc, unsigned int endLoc,
+		std::vector<float>* yArray, float valA, float valB, float fadeInStartLoc);
+	// valA: amp, valB: random number count, curve: seed
+	//float processLineTypeRandom(float xIn, float valA, float valB, float curve, float fadeInStartLoc);
+	std::vector<float> processLineTypeArrayRandom(std::vector<float>* xArray, unsigned int startLoc, unsigned int endLoc,
+		float valA, float valB, float curve, float fadeInStartLoc);
+
+	// updating
+	// adds the points that are
+	// effected by the effector's values changing
+	// ONLY WORKS IN SORTED ARRAYS
+	void getUpdatingFromEffector(std::vector<unsigned int>* updatingPointLocations);
+	// if pointLocation >= 0 -> adds the location to m_needsUpdating
+	// else it will update the whole m_dataArray and m_bakedSamples
+	// changes in the size of m_dataArray (addtition, deletion, ect.)
+	// should to cause a full update
+	void getUpdatingFromPoint(int pointLocation);
+	// adds the points that are changed because their
+	// automation is changed
+	void getUpdatingFromAutomation();
+	// recalculates and sorts m_needsUpdating so
+	// every point is in there only once
+	void getUpdatingOriginals();
+
+	// real getSamples processing
+	void getSamples(unsigned int targetSizeIn, bool* isChangedOut, std::vector<unsigned int>* updatingValuesOut, std::vector<float>* sampleBufferOut);
+	// gets every m_needsUpdating point's line's start and end effector point's location in the effector dataArray
+	// .first = start, .second = line end location (effector dataArray)
+	//void getSamplesLocations(VectorGraphDataArray* effector, std::vector<std::pair<unsigned int, unsigned int>>* effectorDataOut);
+	void getSamplesUpdateLines(VectorGraphDataArray* effector, std::vector<float>* effectorSamples,
+		std::vector<float>* outputXLocations, unsigned int iIn, float stepSize);
```

Comment:
What is `i` in this function?

Reply:
I is m_dataArray[i] (where the points are stored). Will be renamed.